publisher.rb

```diff
@@ -1,25 +1,46 @@
 require 'aws-sdk-sqs'
 require_relative 'load_aws'
 require_relative 'aws_sqs_client'
+require_relative 'q_it_errors'
 
 class Publisher
   
   attr_reader :sqs, :sleep_period, :queue_name
 
   def initialize(options)
     @sqs = AwsSQSClient.new.client
-    @sleep_period = valid_sleep_period?(options[1]) ? options[1].to_i : 5
-    @queue_name = options[0]
+    @queue_name = valid_queue_name(options[0])
```

Comment:
On second thoughts you could have a single `validate` (decide the name) method that in turn calls individual validation methods. Will keep your initialize method clean, with only a single entry point for a validation method for all the arguments of the initialize method.

Reply:
Changing the implementation here.